src/core/logger.c

```diff
@@ -24,16 +13,44 @@ void init_logger(FILE *dst, enum LOGLEVEL level) {
 }
 
 enum LOGLEVEL get_loglevel(const char *level) {
-	int i;
-	for (i = 0; i < NUM_LEVELS; i++) {
-		if (!strcmp(level, loglevels[i])) {
-			return (enum LOGLEVEL) i;
-		}
+	if (strcmp(level, "QUIET") == 0) {
+		return QUIET;
+	} else if (strcmp(level, "ERROR") == 0) {
+		return ERROR;
+	} else if (strcmp(level, "WARN") == 0) {
+		return WARN;
+	} else if (strcmp(level, "INFO") == 0) {
+		return INFO;
+	} else if (strcmp(level, "DEBUG") == 0) {
+		return DEBUG;
+	} else if (strcmp(level, "TRACE") == 0) {
+		return TRACE;
 	}
+
 	fprintf(stderr, "Could not understand log level %s\n", level);
 	abort();
 }
 
+char *print_loglevel(enum LOGLEVEL loglevel) {
+	switch (loglevel) {
+	case QUIET:
+		return "QUIET";
+	case ERROR:
+		return "ERROR";
+	case WARN:
+		return "WARN";
+	case INFO:
+		return "INFO";
+	case DEBUG:
+		return "DEBUG";
+	case TRACE:
+		return "TRACE";
+	}
+	fprintf(stderr, "Could not find log level %d\n", loglevel);
+	abort();
+	return NULL;
```

Comment:
Nit: `abort` is noreturn, no need for this statement.

Reply:
done